core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -75,9 +75,23 @@
      * A response payload has been received. May be called zero or more times depending on whether
      * the call response is empty, a single message or a stream of messages.
      *
-     * @param  payload returned by the server
+     * @param payload returned by the server
+     * @deprecated use {@link #onMessage}
      */
-    public abstract void onPayload(T payload);
+    @Deprecated
+    public void onPayload(T payload) {
+      throw new UnsupportedOperationException();
+    }
+
+    /**
+     * A response message has been received. May be called zero or more times depending on whether
+     * the call response is empty, a single message or a stream of messages.
+     *
+     * @param message returned by the server
+     */
+    public void onMessage(T message) {
+      onPayload(message);
```

Comment:
Just "throw new UnsupportedOperationException();" instead of calling a deprecated method?